rustls/src/client/client_conn.rs

```diff
@@ -326,6 +326,12 @@ impl ClientConfig {
     }
 }
 
+impl AsRef<Arc<CryptoProvider>> for ClientConfig {
```

Comment:
This change isn't mentioned in the commit message and seems unrelated to constructing keys from the crypto provider. Should it be its own commit?

I'm also not sure I understand the motivation vs just using `config.provider` directly. I don't think any other fields of the config structs are exposed with an `AsRef` impl like this. Can you help me understand why the crypto provider field merits special treatment?

Reply:
I think of `AsRef` impls are for "is-a" relations or trivial representational conversions. I'm not sure that `ClientConfig` is a `CryptoProvider`, so maybe this is better as a simple accessor fn?